boa_engine/src/module/mod.rs

```diff
@@ -0,0 +1,734 @@
+//! Boa's implementation of the ECMAScript's module system.
+//!
+//! This module contains the [`Module`] type, which represents an [**Abstract Module Record**][module].
+//! Every module roughly follows the same lifecycle:
+//! - Parse using [`Module::parse`].
+//! - Load all its dependencies using [`Module::load`].
+//! - Link its dependencies together using [`Module::link`].
+//! - Evaluate the module and its dependencies using [`Module::evaluate`].
+//!
```

Comment:
`SimpleModuleLoader` should probably be mentioned earlier in the header doc, right? 

Something like:
```suggestion
//! This module contains the [`Module`] type, which represents an 
//! [**Abstract Module Record**][module], a `ModuleLoader` trait
//! for custom module loader implementations, and a default 
//! `SimpleModuleLoader`.
//!
//! Every module roughly follows the same lifecycle:
//! - Parse using [`Module::parse`].
//! - Load all its dependencies using [`Module::load`].
//! - Link its dependencies together using [`Module::link`].
//! - Evaluate the module and its dependencies using [`Module::evaluate`].
//!
```
Then we can expand on the topics below like you've currently done. 😄